test/github-issues/3288/issue-3288.ts

```diff
@@ -0,0 +1,34 @@
+import "reflect-metadata";
+import { closeTestingConnections, createTestingConnections, reloadTestingDatabases } from "../../utils/test-utils";
+import { Connection } from "../../../src/connection/Connection";
+import { expect } from "chai";
+import { User } from "./entity/user.entity";
+import { Photo } from "./entity/photo.entity";
+
+describe("github issues > #3288 null foreign key when inserting with number instead of relation object", () => {
+
+    let connections: Connection[];
+    before(async () => connections = await createTestingConnections({
+        entities: [__dirname + "/entity/*{.js,.ts}"],
+    }));
+    beforeEach(() => reloadTestingDatabases(connections));
+    after(() => closeTestingConnections(connections));
+
+    it("should assign foreign field key to relation id", () => Promise.all(connections.map(async connection => {
+        const user = new User();
+        
+        const dbUser = await connection.manager.save(user);
+
+        const photo = new Photo();
+        photo.user = dbUser.id as any;
```

Comment:
you don't feel like ` as any` is something wrong? You should simply do `photo.user = dbUser`